src/test/java/org/wildfly/security/sasl/gs2/Gs2Test.java

```diff
@@ -524,54 +614,44 @@ private SaslClient getSaslClient(final String[] mechanisms, final String authori
         }
     }
 
-    private class Gs2ServerCallbackHandler implements CallbackHandler {
+    //TODO I don't like the indirect tests having to pass in a callback handler to get this information
+    private static class IndirectCallbackHandler implements CallbackHandler {
         private final String bindingType;
         private final byte[] bindingData;
 
-        public Gs2ServerCallbackHandler(final String bindingType, final byte[] bindingData) {
+        private IndirectCallbackHandler(String bindingType, byte[] bindingData) {
             this.bindingType = bindingType;
             this.bindingData = bindingData;
         }
 
+        @Override
         public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
             for (Callback callback : callbacks) {
                 if (callback instanceof ChannelBindingCallback) {
```

Comment:
If we need to discuss something further just create a critical Jira issue against one of the upcoming 1.0.x releases and we can revisit later rather than blocking the PR.